spec/services/process_submission_service_spec.rb

```diff
@@ -0,0 +1,483 @@
+  let(:downloaded_body_parts) { mock_downloaded_files }
+  let(:body_part_content) do
+    {
+      'text/plain' => 'some plain text',
+      'text/html' => '<html>some html</html>'
+    }
+  end
+  let(:token) { 'some token' }
+  let(:headers) { {'x-encrypted-user-id-and-token' => token} }
+
+  describe '#perform' do
+    let(:submission_detail) do
+      {
+        'from' => 'some.one@example.com',
+        'to' => 'destination@example.com',
+        'subject' => 'mail subject',
+        'type' => 'email',
+        'body_parts' => {
+          'text/html' => 'https://tools.ietf.org/html/rfc2324',
+          'text/plain' => 'https://tools.ietf.org/rfc/rfc2324.txt'
+        },
+        'attachments' => [
+          {
+            'type' => 'output',
+            'mimetype' => 'application/pdf',
+            'url' => '/api/submitter/pdf/default/guid1.pdf',
+            'filename' => 'form1'
+          },
+          {
+            'type' => 'output',
+            'mimetype' => 'application/pdf',
+            'url' => '/api/submitter/pdf/default/guid2.pdf',
+            'filename' => 'form2'
+          }
+        ]
+      }
+    end
+
+    let(:processed_attachments) do
+      [
+        Attachment.new({
+          path: '/path/to/file1',
+          type: 'output',
+          mimetype: 'application/pdf',
+          url: 'http://service-slug.formbuilder-services-test:3000/api/submitter/pdf/default/guid1.pdf',
+          filename: 'form1'
+        })
+      ]
+    end
+
+    let(:submission) do
+      Submission.create!(
+        encrypted_user_id_and_token: token,
+        status: 'queued',
+        submission_details: [submission_detail],
+        service_slug: 'service-slug'
+      )
+    end
+
+    let(:detail_objects) do
+      [EmailSubmissionDetail.new(submission_detail)]
+    end
+
+    let(:urls) do
+      ['http://service-slug.formbuilder-services-test:3000/api/submitter/pdf/default/guid1.pdf', 'http://service-slug.formbuilder-services-test:3000/api/submitter/pdf/default/guid2.pdf']
+    end
+
+    let(:mock_send_response){ {'key' => 'send response'} }
+
+    before do
+      allow(EmailService).to receive(:send_mail).and_return(mock_send_response)
+      allow(DownloadService).to receive(:download_in_parallel).and_return(
+        mock_downloaded_files
+      )
+      allow(subject).to receive(:retrieve_mail_body_parts).and_return(body_part_content)
+    end
+
+    context 'given a valid submission_id' do
+      let(:submission_id) { submission.id }
+
+      before do
+        allow(Submission).to receive(:find).with(submission_id).and_return(submission)
+      end
+
+      it 'loads the Submission' do
+        expect(Submission).to receive(:find).with(submission_id).and_return(submission)
```

Comment:
Line 99 and Line 103 just test each other. Line 99 will blow up if it doesn't receive find with the submission id, so I would say that either the L102 spec doesn't need to exist or we can put the `before`/`do` block after L102 in its own context.

Reply:
I've made a slight tweak and removed the extra test